gpu-simulator/trace-parser/trace_parser.h

```diff
@@ -75,6 +75,30 @@ struct inst_trace_t {
   ~inst_trace_t();
 };
 
+class PipeReader {
+ public:
+  PipeReader() {}
+  void OpenFile(const std::string &filePath);
```

Comment:
For more robust C++ code, I would suggest implementing the RAII (Resource Acquisition is Initialization) principle. That is, merge the functionality of `OpenFile` into the constructor of this class and call `popen` there. The destructor is where you `pclose`, which is already there and good. 

We are already allocating the `kernel_trace_t` object and calling `OpenFile` back to back, so the change should be minimal.